android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java

```diff
@@ -646,15 +646,18 @@ public void updateBalanceMaybeSwap() {
         updateBalanceMaybeSwap(getCurrentSelectedAccountAddr());
     }
 
-    private void populateBalance(String balance, boolean from) {
+    private void populateBalance(String balance, boolean from, int responseDecimals) {
         BlockchainToken token = from ? mCurrentBlockchainToken : mCurrentSwapToBlockchainToken;
         TextView textView = from ? mFromBalanceText : mToBalanceText;
 
         double fromToBalance = 0;
 
         // some old calls are returning ETH result when SOL is selected so do nothing
         try {
-            int decimals = token != null ? token.decimals : Utils.ETH_DEFAULT_DECIMALS;
+            int decimals = responseDecimals;
+            if (decimals == 0) {
```

Comment:
Maybe change the name to `SOLDecimals`? I'm not sure what `response` means here. Also probably using `null` or probably at lesat -1 is better than 0, we have ETH NFTs which also has decimals as 0, this can confuse people in the future.

Reply:
Or even better, `int decimals = mSelectedNetwork.coin == CoinType.SOL ? responseDecimals : token != null ? token.decimals : Utils.ETH_DEFAULT_DECIMALS;` ?

Reply:
it's not sol, it's used for SPL tokens and a call to get an SPL token balance returns `decimals` which we should use to calculate a balance correctly
```
mJsonRpcService.getSplTokenAccountBalance(address, blockchainToken.contractAddress,
                    mSelectedNetwork.chainId,
                    (amount, decimals, uiAmountString, solanaProvideError, error_message) -> {
```
Solana NFT uses 0 as decimals for tokens, I don't think changing to -1 is a good idea, why the current logic doesn't work? we pass `decimals` from a response if it exists or otherwise 0 and we have an old logic if it's 0. I think -1 will bring more confusion here.

Reply:
> Or even better, `int decimals = mSelectedNetwork.coin == CoinType.SOL ? responseDecimals : token != null ? token.decimals : Utils.ETH_DEFAULT_DECIMALS;` ?

that doesn't apply for all Solana assets a native SOL doesn't use `responseDecimals`